bin/testcase.py

```diff
@@ -231,17 +232,15 @@ def validate_format(
             flags = args if flags is None else flags + args
 
             ret = validator.run(self, mode=mode, constraints=constraints, args=flags)
-            if ret.ok is not True and ret.ok != config.RTV_WA:
-                bar.log(f"Expected exit code {config.RTV_AC} or {config.RTV_WA}, got {ret.ok}")
-                ret.ok = config.RTV_WA
-            ok = ret.ok == True
+            if ret.ok == ExecStatus.ERROR:
+                bar.log(f"Unxpected exit code!")
```

Comment:
We could still print the exit code here (and perhaps even still mention that it should be either `config.RTV_AC` or `config.RTV_WA`):
```suggestion
                bar.log(f'Expected exit code {config.RTV_AC} or {config.RTV_WA}, got ${ret.returncode}')
```

---

Also, it's somewhat unfortunate that [Python uses `returncode`](https://docs.python.org/3/library/subprocess.html#subprocess.CompletedProcess.returncode) as name for what I know as the "exit code" of a program... I wanted to propose renaming `ExecResult.returncode` to `.exit_code`, but on the Python side, it's probably best to stay consistent with the standard library. However, in user-facing messages, I'm in favour of writing "exit code" :slightly_smiling_face:

Reply:
we can display the actual exit code but we should not display what we expect since that changes depending on the type of validator?

Reply:
Fair point, this looks good to me then :smile: